### PR TITLE
email,url,ip format  are now recognized 

### PR DESCRIPTION
**Pull request by ismnoiet**

### CODE REVIEW

**Comment by mohsen1** [comment]
Please update files in src folder instead of dist and then run `npm run browserify` to make a new build.

**Comment by ismnoiet** [comment]
ok  i got it
I have pushed a second commit which respect what you've said exactly

File: src/index.js [hunk 1]
```diff
@@ -18,6 +26,20 @@ function convert (json, options = {}) {
     return {};
   }
 
+  // check for a URL, before testing against the type "string",
+  // because every URL is also a string 
+  if(regex.url.test(json)){
+    return { type: 'url'};
```
Comment (mohsen1) [inline src/index.js]: All of these tests should go after `if (typeof json === 'string') {` and when the if condition goes through it should return `{type: 'string', format: 'url'}`.

File: src/index.js [hunk 2]
```diff
@@ -2,6 +2,14 @@
 
 import isEqual from 'lodash.isequal';
 
+// Regex used here are all from 
+// http://code.tutsplus.com/tutorials/8-regular-expressions-you-should-know--net-6149 
+var regex = {
```
Comment (mohsen1) [inline src/index.js]: use `const` and rename `regex` to `FORMAT_REGEX`

**Comment by ismnoiet** [comment]
@mohsen1  Can we have the case where 
type = 'string' 
format = 'string' ?

**Comment by mohsen1** [comment]
No, these are valid types: http://json-schema.org/latest/json-schema-core.html#anchor8

You should also add a bunch of test for those types in `primitives.js` test file.

**Comment by ismnoiet** [comment]
I mean if we have the type string and the format is not in our list 
example 
`{username:'hamza'}`
in this case what is the format of username attribute ?

**Comment by mohsen1** [comment]
Yes, if it doesn't match it will simply return `{type: 'string'}`.

File: src/index.js [hunk 3]
```diff
@@ -21,6 +29,18 @@ function convert (json, options = {}) {
   // primitives
   if (typeof json === 'string') {
 
+    if(FORMAT_REGEX.url.test(json)){
+      return { type:'string',format : 'url'};
```
Comment (mohsen1) [inline src/index.js]: Please add space after the comma and not after opening brace:

```
return {type:'string', format : 'url'};
```

**Comment by mohsen1** [comment]
@ismnoiet can you add three simple tests for this so I can merge it?

**Comment by ismnoiet** [comment]
@mohsen1 wait for 10 min and i'll commit  the tests

File: src/index.js [hunk 4]
```diff
@@ -21,6 +29,18 @@ function convert (json, options = {}) {
   // primitives
   if (typeof json === 'string') {
 
+    if(FORMAT_REGEX.url.test(json)){
```
Comment (mohsen1) [inline src/index.js]: Please also add a space after `if`. Sorry for nitpicking!

**Comment by mohsen1** [comment]
After you've done please rebase into one commit.

**Comment by ismnoiet** [comment]
@mohsen1 I think i'm done with this, hopefully didn't forget something :+1:

**Comment by mohsen1** [comment]
Can you rebase into one commit?

```
git rebase -i HEAD~7
```

**Pull request closed without merging**